### PR TITLE
Update HTTP Security Response Headers Cheat Sheet

### PR DESCRIPTION
**Pull request by fcanogab**
Update HTTP Security Response Headers Cheat Sheet to provide to developers up-to-date information about when a header is recommended and the impact of using.

Thank you for submitting a Pull Request (PR) to the Cheat Sheet Series. 

> :triangular_flag_on_post: If your PR is related to grammar/typo mistakes, please double-check the file for other mistakes in order to fix all the issues in the current cheat sheet.

Please make sure that for your contribution:

- [x] In case of a new Cheat Sheet, you have used the [Cheat Sheet template](https://github.com/OWASP/CheatSheetSeries/blob/master/templates/New_CheatSheet.md).
- [x] All the markdown files do not raise any validation policy violation, see the [policy](https://github.com/OWASP/CheatSheetSeries/actions?query=workflow%3A%22Markdown+Link+Check%22).
- [x] All the markdown files follow these [format rules](https://github.com/OWASP/CheatSheetSeries/blob/master/CONTRIBUTING.md#markdown).
- [x] All your assets are stored in the **assets** folder.
- [x] All the images used are in the **PNG** format.
- [x] Any references to websites have been formatted as [TEXT](URL)
- [x] You verified/tested the effectiveness of your contribution (e.g., the defensive code proposed is really an effective remediation? Please verify it works!).
- [x] The CI build of your PR pass, see the build status [here](https://github.com/OWASP/CheatSheetSeries/actions).

If your PR is related to an issue, please finish your PR text with the following line:

This PR covers issue #991 

Thank you again for your contribution :smiley:

### CODE REVIEW

**Comment by righettod** [comment]
Hi,

Feel free to also make a ref to the [OWASP Secure Headers Project](https://owasp.org/www-project-secure-headers/) 😃 

Dom

**Comment by mwjcomputing** [comment]
This looks great. Awesome job.

File: cheatsheets/HTTP_Headers_Cheat_Sheet.md [hunk 1]
```diff
@@ -10,32 +10,44 @@ In this cheat sheet, we will review all security-related HTTP headers, recommend
 
 ### X-Frame-Options
 
-The `X-Frame-Options` HTTP response header can be used to indicate whether or not a browser should be allowed to render a page in a `<frame>`, `<iframe>`, `<embed>` or `<object>`. Sites can use this to avoid clickjacking attacks, by ensuring that their content is not embedded into other sites.
+The `X-Frame-Options` HTTP response header can be used to indicate whether or not a browser should be allowed to render a page in a `<frame>`, `<iframe>`, `<embed>` or `<object>`. Sites can use this to avoid [clickjacking](https://owasp.org/www-community/attacks/Clickjacking) attacks, by ensuring that their content is not embedded into other sites.
+
+Content Security Policy (CSP) frame-ancestors directive obsoletes X-Frame-Options for supporting browsers [source](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Frame-Options).
```
Comment (szh) [inline cheatsheets/HTTP_Headers_Cheat_Sheet.md:15]:
```suggestion
Content Security Policy (CSP) frame-ancestors directive obsoletes X-Frame-Options for supporting browsers ([source](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Frame-Options)).
```

File: cheatsheets/HTTP_Headers_Cheat_Sheet.md [hunk 2]
```diff
@@ -10,32 +10,44 @@ In this cheat sheet, we will review all security-related HTTP headers, recommend
...
 Please see [Mozilla X-XSS-Protection](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-XSS-Protection) for details.
 
 ### X-Content-Type-Options
 
-The `X-Content-Type-Options` response HTTP header is used by the server to prevent browsers from guessing the media type ( MIME type).
-This is known as **MIME sniffing** in which the browser guesses the correct MIME type by looking at the contents of the resource.
-The absence of this header might cause browsers to transform non-executable content into executable content.
+The `X-Content-Type-Options` response HTTP header is used by the server to indicate to the browsers that the MIME types advertised in the Content-Type headers should be followed and not guessed.
+
+This header is used to block browsers [MIME type sniffing](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types#mime_sniffing), which can transform non-executable MIME types into executable MIME types ([MIME Confusion Attacks](https://blog.mozilla.org/security/2016/08/26/mitigating-mime-confusion-attacks-in-firefox/)).
```
Comment (szh) [inline cheatsheets/HTTP_Headers_Cheat_Sheet.md:45]:
```suggestion
This header is used to block browsers' [MIME type sniffing](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types#mime_sniffing), which can transform non-executable MIME types into executable MIME types ([MIME Confusion Attacks](https://blog.mozilla.org/security/2016/08/26/mitigating-mime-confusion-attacks-in-firefox/)).
```

File: cheatsheets/HTTP_Headers_Cheat_Sheet.md [hunk 3]
```diff
@@ -10,32 +10,44 @@ In this cheat sheet, we will review all security-related HTTP headers, recommend
...
 
 #### Recommendation
 
+Set the Content-Type header correctly through the site.
```
Comment (szh) [inline cheatsheets/HTTP_Headers_Cheat_Sheet.md:49]:
```suggestion
Set the Content-Type header correctly throughout the site.
```

File: cheatsheets/HTTP_Headers_Cheat_Sheet.md [hunk 4]
```diff
@@ -65,59 +79,85 @@ The `Content-Type` representation header is used to indicate the original media
...
 
 The `Expect-CT` header lets sites opt-in to reporting of Certificate Transparency (CT) requirements. Given that mainstream clients now require CT qualification, the only remaining value is reporting such occurrences to the nominated report-uri value in the header. The header is now less about enforcement and more about detection/reporting.
 
-Please note Mozilla states that [this header will be obsolete](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Expect-CT) in June 2021.
-
 #### Recommendation
 
-Set Certificate Transparency so user agents report Expect-CT failures.
-> `Expect-CT: max-age=604800, report-uri="https://foo.example/report"`
+Not use it. Mozilla [recommends](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Expect-CT) avoiding it, and removing it from existing code if possible.
```
Comment (szh) [inline cheatsheets/HTTP_Headers_Cheat_Sheet.md:106]:
```suggestion
Do not use it. Mozilla [recommends](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Expect-CT) avoiding it, and removing it from existing code if possible.
```

File: cheatsheets/HTTP_Headers_Cheat_Sheet.md [hunk 5]
```diff
@@ -65,59 +79,85 @@ The `Content-Type` representation header is used to indicate the original media
...
+
+### Content-Security-Policy (CSP)
 
-### Content-Security-Policy
+Content Security Policy (CSP) is a security feature that is used to specify the origin of content that is allowed to be loaded on a website or in a web applicationis. It is an added layer of security that helps to detect and mitigate certain types of attacks, including Cross-Site Scripting (XSS) and data injection attacks. These attacks are used for everything from data theft to site defacement to distribution of malware.
```
Comment (szh) [inline cheatsheets/HTTP_Headers_Cheat_Sheet.md:110]:
```suggestion
Content Security Policy (CSP) is a security feature that is used to specify the origin of content that is allowed to be loaded on a website or in a web applications. It is an added layer of security that helps to detect and mitigate certain types of attacks, including Cross-Site Scripting (XSS) and data injection attacks. These attacks are used for everything from data theft to site defacement to distribution of malware.
```

File: cheatsheets/HTTP_Headers_Cheat_Sheet.md [hunk 6]
```diff
@@ -65,59 +79,85 @@ The `Content-Type` representation header is used to indicate the original media
...
 
 #### Recommendation
 
-Content Security Policy is very complex to configure and maintain. For an explanation on customization options, please read [Content Security Policy Cheat Sheet](Content_Security_Policy_Cheat_Sheet.md)
+Content Security Policy is complex to configure and maintain. For an explanation on customization options, please read [Content Security Policy Cheat Sheet](Content_Security_Policy_Cheat_Sheet.md)
 
 ### Access-Control-Allow-Origin
 
-The `Access-Control-Allow-Origin` response header indicates whether the response can be shared with requesting code from the given origin.
+If you don't use this header, your site is protected by default by the Same Origin Policy (SOP). What this header does is relaxing this control in specified circumstances.
```
Comment (szh) [inline cheatsheets/HTTP_Headers_Cheat_Sheet.md:120]:
```suggestion
If you don't use this header, your site is protected by default by the Same Origin Policy (SOP). What this header does is relax this control in specified circumstances.
```

**Review by szh — CHANGES_REQUESTED** [review]
I didn't have time to review the content thoroughly yet, but I added some suggestions for spelling and grammar.

**Review by jmanico — APPROVED** [review]

**Review by jmanico — CHANGES_REQUESTED** [review]
Use a Content Security Policy (CSP) that disables the use of inline JavaScript ('unsafe-inline').

unsafe-inline actually disables this protection, can you please remove that portion?

**Pull request merged**